### DISCUSSION TITLE
Difficulty to Install on MacOS -- assist requested

### DISCUSSION THREAD

**Discussion by GWhized** [vkbo/novelWriter#600] [Help]
Hi,
{I'm new to local builds from GitHub -- assist requested}
I'm having trouble building on MacOS.  Same failure on virtual.
I have done these steps successfully:
√ downloaded ZIP of novelWriter-main and unzip'd to a user subdirectory = ../novelWriter-main
√ Create virtual environment:
    √ python3 -m venv novelWriter-main/
    √ source novelWriter-main/bin/activate
√ python3 --version = Python 3.7.6
√ pip --version   =  pip 19.2.3
but, when I "pip3 install --user -r requirements.txt" I get these errors:
Looking in indexes: https://pypi.apple.com/simple
Collecting pyqt5>=5.2.1 (from -r requirements.txt (line 1))
  WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x10b823a90>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /simple/pyqt5/
  ERROR: Could not find a version that satisfies the requirement pyqt5>=5.2.1 (from -r requirements.txt (line 1)) (from versions: none)
ERROR: No matching distribution found for pyqt5>=5.2.1 (from -r requirements.txt (line 1))

Similar failure on simple "pip install novelwriter"
Looking in indexes: https://pypi.apple.com/simple
Collecting novelwriter
  WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x10b23ea90>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known')': /simple/novelwriter/
  ERROR: Could not find a version that satisfies the requirement novelwriter (from versions: none)
ERROR: No matching distribution found for novelwriter

**Comment by vkbo** [comment]
I'm not an expert on Mac, but it seems to me you're not checking the official PyPi repository, but some apple substitute. The clue is in the line:
`Looking in indexes: https://pypi.apple.com/simple`

As far as I understand, there's a difference between the Python toolset that comes with macOS, and the one provided by Brew. Apple seems to inject itself into the flow in the former. I don't know much about what it is they do there. The install instructions are designed for use with Brew. See https://novelwriter.readthedocs.io/en/stable/setup_mac.html

Ideally, there should be a proper installer for Mac, but I don't have direct access to a Mac, so I require outside help to provide a better way of installing novelWriter on that OS.

Edit: Updated link

**Comment by leej3** [comment]
I just got it working on Mac OS (M1, Apple Silicon). I suspect this will work with an Intel Mac too though. It was quite the challenge. I tried various approaches but ultimately what I got working was using home-brew in combination with homebrew's pip:


```
brew install pyqt@5 enchant

# "which pip3" should show you pip3 installed by homebrew... i.e. /opt/homebrew/bin/pip3 or something like that
# use homebrew's pip to install lxml:
pip3 install lxml

# now you can use homebrew's python to run novelWriter. Assuming you are in the novelWriter repository/directory:
python3 novelWriter.py
```

**Reply by vkbo** [reply]
Yes, as the [documentation states](https://novelwriter.readthedocs.io/en/stable/setup_mac.html), you need to install Python and pip from homebrew.

**Comment by leej3** [comment]
It’s the pyqt that I was struggling with. Pip/conda didn’t help me. Installing it with the ***@***.***” notation with homebrew did the trick. This very well may just be a temporary thing until more widespread support is established for the new apple chip architecture. I figured I’d share in case it helps the next poor unfortunate. Thanks for the great package. Works very nicely.  

John

**Reply by vkbo** [reply]
Ah, perhaps I misunderstood.

I know from colleagues using Python on macOS that there are some quirks with homebrew vs system Python, which on testing also affected PyQt5, but to be honest I know very little about Macs aside from what it has in common with Linux.

Feel free to make a PR to the documentation if you think there's something that can be improved. It would be greatly appreciated. What is there is basically just what I figured out when getting novelWriter to run in a virtual machine running Catalina.

The docs are in the `docs` folder of the main repo.

**Reply by leej3** [reply]
I had a chance to look into this on a mac with an intel cpu: the docs work as expected. I think the only issue with the OP is "pip install novelWriter" does not have the correct letter case for the "W".

Regarding M1 macs, It is indeed confusing. As described above it is more difficult to get something working with pip/conda because wheels have not yet been packaged for the chip architecture. This means that you cannot use software environment solutions but instead have to use brew (along with its pip) to manage all dependencies (and so use the commands python3 and pip3). I'm reluctant to document the hacky solution formally since your current docs work on the typical mac and will have resolved itself in the coming months as more software devs out there get access to the new architecture.

**Reply by vkbo** [reply]
I don't think pip is case sensitive. I've tried to keep everything in lower case on the file level though.

As for the issue with wheels, all the dependencies need binary builds, so I hope it will be resolved. I'm very happy now that I made an effort to keep the dependency list as short as possible.

**Reply by leej3** [reply]
> I don't think pip is case sensitive.

Huh. Just tested it. That's a new one for me.  

>  I'm very happy now that I made an effort to keep the dependency list as short as possible.

Yep, harder for the developer, much easier for the users :)
